shopinvader_sale_coupon/services/customer.py

```diff
@@ -12,35 +12,52 @@ def _parser_coupon(self):
         return [
             "id",
             "code",
-            "state",
             "expiration_date",
             (
                 "program_id:program",
                 [
                     "id",
                     "name",
-                    "reward_type",
-                    "reward_product_id",
-                    "reward_product_quantity",
-                    "discount_type",
-                    "discount_percentage",
-                    "discount_fixed_amount",
+                    "program_type",
+                    ("reward_ids:rewards", self._parser_coupon_reward()),
```

Comment:
As the programs & rewards model structures changed a lot, I don't know how to use the parser to return the same data structure, preserving the old field names.